src/core/annotation.js

```diff
@@ -451,46 +454,59 @@ class Annotation {
     }
   }
 
-  /**
-   * Set the color and take care of color space conversion.
-   * The default value is black, in RGB color space.
-   *
-   * @public
-   * @memberof Annotation
-   * @param {Array} color - The color array containing either 0
-   *                        (transparent), 1 (grayscale), 3 (RGB) or
-   *                        4 (CMYK) elements
-   */
-  setColor(color) {
-    const rgbColor = new Uint8ClampedArray(3);
+  _getColor(color, defaultColor) {
     if (!Array.isArray(color)) {
-      this.color = rgbColor;
-      return;
+      return defaultColor;
     }
 
+    const rgbColor = defaultColor || new Uint8ClampedArray(3);
+
     switch (color.length) {
       case 0: // Transparent, which we indicate with a null value
-        this.color = null;
-        break;
-
+        return null;
       case 1: // Convert grayscale to RGB
         ColorSpace.singletons.gray.getRgbItem(color, 0, rgbColor, 0);
-        this.color = rgbColor;
-        break;
+        return rgbColor;
 
       case 3: // Convert RGB percentages to RGB
         ColorSpace.singletons.rgb.getRgbItem(color, 0, rgbColor, 0);
-        this.color = rgbColor;
-        break;
+        return rgbColor;
 
       case 4: // Convert CMYK to RGB
         ColorSpace.singletons.cmyk.getRgbItem(color, 0, rgbColor, 0);
-        this.color = rgbColor;
-        break;
+        return rgbColor;
 
       default:
-        this.color = rgbColor;
-        break;
+        return defaultColor;
+    }
+  }
+
+  /**
+   * Set the color and take care of color space conversion.
+   * The default value is black, in RGB color space.
+   *
+   * @public
+   * @memberof Annotation
+   * @param {Array} color - The color array containing either 0
+   *                        (transparent), 1 (grayscale), 3 (RGB) or
+   *                        4 (CMYK) elements
+   */
+  setColor(color) {
+    this.color = this._getColor(color, new Uint8ClampedArray(3));
+  }
+
+  /**
+   * Set the color for background and border if any.
+   * The default values are transparent.
+   *
+   * @public
+   * @memberof Annotation
+   * @param {Dict}  - The MK dictionary
+   */
+  setBorderAndBackgroundColors(mk) {
+    if (mk) {
```

Comment:
*Drive-by comment, to prevent bugs in corrupt documents:*
```suggestion
    if (mk instanceof Dict) {
```

Reply:
You're so right...